atom/browser/native_window_mac.mm

```diff
@@ -1351,6 +1362,83 @@ static bool FromV8(v8::Isolate* isolate, v8::Handle<v8::Value> val,
 void NativeWindowMac::SetSkipTaskbar(bool skip) {
 }
 
+void NativeWindowMac::SetSimpleFullScreen(bool simple_fullscreen) {
+  NSWindow* window = GetNativeWindow();
+
+  if (simple_fullscreen && !is_simple_fullscreen_) {
+    is_simple_fullscreen_ = true;
+
+    // Take note of the current window size
+    original_frame_ = [window frame];
+
+    simple_fullscreen_options_ = [NSApp currentSystemPresentationOptions];
+
+    // We can simulate the pre-Lion fullscreen by auto-hiding the dock and menu bar
+    NSApplicationPresentationOptions options =
+        NSApplicationPresentationAutoHideDock +
+        NSApplicationPresentationAutoHideMenuBar;
+    [NSApp setPresentationOptions:options];
+
+    was_maximized_ = IsMaximized();
+    was_minimizable_ = IsMinimizable();
+    was_maximizable_ = IsMaximizable();
+    was_resizable_ = IsResizable();
+    was_movable_ = IsMovable();
+
+    // if (!was_maximized_) Maximize();
+
+    NSRect fullscreenFrame = [window.screen frame];
+
+    if ( !fullscreen_window_title() ) {
+      // Hide the titlebar
+      SetStyleMask(false, NSTitledWindowMask);
+
+      // Resize the window to accomodate the _entire_ screen size
+      fullscreenFrame.size.height -= [[[NSApplication sharedApplication] mainMenu] menuBarHeight];
+    } else {
+      // No need to hide the title, but we should still hide the window buttons
+      [[window standardWindowButton:NSWindowZoomButton] setHidden:YES];
+      [[window standardWindowButton:NSWindowMiniaturizeButton] setHidden:YES];
+      [[window standardWindowButton:NSWindowCloseButton] setHidden:YES];
+    }
+
+    [window setFrame:fullscreenFrame display: YES animate: YES];
+
+    // Fullscreen windows can't be resized, minimized, etc.
+    if (was_minimizable_) SetMinimizable(false);
```

Comment:
The window states are just style masks, I think you can just store the original window style masks instead of separate window states.

Reply:
That makes a lot of sense, thanks for the feedback!